src/test/java/org/mifos/integrationtest/cucumber/stepdef/BillPayStepDef.java

```diff
@@ -67,7 +67,7 @@ public void iCallTheGetBillsApiWithBillidWithExpectedStatusOf(int expectedStatus
         requestSpec.queryParam("fields", "inquiry");
         scenarioScopeState.response = RestAssured.given(requestSpec).baseUri(billPayConnectorConfig.billPayContactPoint).expect()
                 .spec(new ResponseSpecBuilder().expectStatusCode(expectedStatus).build()).when()
-                .get(billPayConnectorConfig.inquiryEndpoint.replace("{billId}", billId)).andReturn().asString();
+                .get(billPayConnectorConfig.inquiryEndpoint.replace("{billId}", BaseStepDef.billId)).andReturn().asString();
```

Comment:
Please check the latest code change, We have discontinued usage of static variables in Integration tests to make it threadsafe. Please use ScenarioStatue object for sharing state between steps.

Reply:
this was kept as such even in dhruv's code, so left it as such

Reply:
Let me check this